packages/utils/telemetry-utils/src/mockLogger.ts

```diff
@@ -20,20 +20,24 @@ import { ITelemetryLoggerExt, ITelemetryPropertiesExt } from "./telemetryTypes.j
  * @alpha
  */
 export class MockLogger implements ITelemetryBaseLogger {
-	events: ITelemetryBaseEvent[] = [];
+	private _events: ITelemetryBaseEvent[] = [];
 
-	constructor(public readonly minLogLevel?: LogLevel) {}
+	public get events(): readonly ITelemetryBaseEvent[] {
```

Comment:
These events were previously externally mutable. Updates this to use a private field to track local state and returns a readonly view of the data in the public getter.

Reply:
To really be safe, and have a sane contract here, we should really return a readonly copy of the data, and document that behavior.

Reply:
Unfortunately, we have tests depending on this mutability. I've reverted this change for now and left a TODO to clean this up later.